libraries/plugins/asset_storage/asset_storage_api.cpp

```diff
@@ -0,0 +1,97 @@
+#include <futurepia/asset_storage/asset_storage_api.hpp>
+#include <futurepia/app/state.hpp>
+
+#include <functional>
+
+namespace futurepia { namespace asset_storage {
+
+   namespace detail {
+      class asset_api_impl
+      {
+         public:
+            asset_api_impl( futurepia::app::application& app ):_app( app ) {}
+
+            optional< asset_api_object > get_asset_by_name( string asset_name ) const;
+            vector< asset_api_object > get_assets_by_owner( string owner ) const;
+            vector< asset_event_api_object > get_asset_events( string asset_name ) const;
+
+            futurepia::chain::database& database() { return *_app.chain_database(); }
+
+         private:
+            futurepia::app::application& _app;
+      };
+
+      optional< asset_api_object > asset_api_impl::get_asset_by_name( string asset_name ) const
+      {
+         const auto& asset_idx = _app.chain_database()->get_index< asset_object_index >().indices().get< by_name >();
```

Comment:
Is it possible that chain_database returns 0 or empty pointer?

Reply:
The object that is referenced is always created with the main application class and it is deleted only with the application (node) itself. So there are no checks to prevent it, but it should not return 0.

The value was not checked in Graphene, too.